lib/griddler/cloudmailin/adapter.rb

```diff
@@ -13,22 +13,42 @@ def self.normalize_params(params)
       end
 
       def normalize_params
-        {
-          to: params[:envelope][:to].split(','),
+        normalized_params = {
+          to: tos,
           cc: ccs,
-          from: params[:envelope][:from],
+          from: params[:headers][:From],
           subject: params[:headers][:Subject],
           text: params[:plain],
-          attachments: params.fetch(:attachments) { [] },
+          attachments: params.fetch(:attachments) { {} }.values,
         }
+
+        normalized_params[:bcc] = bcc unless bcc.empty?
+        normalized_params
       end
 
       private
 
       attr_reader :params
 
+      def recipients(field)
+        params[:headers][field].to_s.split(',').map(&:strip)
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.